### PR TITLE
feat: additional parameters, including CMK, for `avm/res/net-app/net-app-account`

### PR DESCRIPTION
**Pull request by bobmclane999**
## Pipeline Reference

| Pipeline |
| -------- |
|  [avm.res.net-app.net-app-account](https://github.com/bobmclane999/bicep-registry-modules/actions/workflows/avm.res.net-app.net-app-account.yml)       |

See this run: https://github.com/bobmclane999/bicep-registry-modules/actions/runs/9418453527

## Type of Change

- [ ] Update to CI Environment or utlities (Non-module effecting changes)
- [ ] Azure Verified Module updates:
  - [ ] Bugfix containing backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [ ] I'm sure there are no other open Pull Requests for the same update/change
- [ ] I have run `Set-AVMModule` locally to generate the supporting module files.
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

**Comment by bobmclane999** [comment]
@microsoft-github-policy-service agree [company="Linklaters"]

File: avm/res/net-app/net-app-account/main.bicep [hunk 1]
```diff
@@ -33,6 +36,21 @@ param managedIdentities managedIdentitiesType
 @description('Optional. Array of role assignments to create.')
 param roleAssignments roleAssignmentType
 
+@description('Optional. The key name to use for encryption.')
```
Comment (eriqua) [inline avm/res/net-app/net-app-account/main.bicep:39]: Hey @bobmclane999 thanks for this contribution. 
Please be aware in general that when it comes to CMK encryption we should leverage a common interface as detailed by AVM specs: https://azure.github.io/Azure-Verified-Modules/specs/shared/interfaces/#customer-managed-keys

In addition, a few months ago the implementation was not completed due to the NetApp encryption feature not being GA and additional requirement to enable a prerequisite at subscription scope. Ref CARML issue https://github.com/Azure/ResourceModules/issues/2991
cc @AlexanderSehr, @ahmadabdalla and last but not least the module owner @fbinotto to confirm

Comment (bobmclane999) [reply avm/res/net-app/net-app-account/main.bicep:39]: Hi @eriqua  no problem this is my 1st attempt or a PR to you guys after a great Teams chat the other day with @jtracey93 so please be gentle with me :)

Have successfully managed to deploy with CMK and according to the below link it appears to be GA also have registered on our Non Prod subscription where I've been testing the code and appears to have registered so fingers crossed  - 
image
 
https://learn.microsoft.com/en-us/azure/azure-netapp-files/configure-customer-managed-keys?tabs=azure-portal

I see so on the CMK is that something you or Jack (@jtracey93 ) can help me with?

Kind Regards
Shaun @bobmclane999

Comment (jtracey93) [reply avm/res/net-app/net-app-account/main.bicep:39]: Hey @bobmclane999,

Firstly, thanks for the PR here. Ive just had a look and I think @eriqua  is saying we should use the AVM spec and use the User Defined Type and standard shared interface as we mandate in AVM resource modules for consumer consistency.

I have put in a PR (https://github.com/bobmclane999/bicep-registry-modules/pull/1) to your PR branch on your fork to do this, so you can see the changes, if you are happy please approve and merge them into your branch, which will then update this PR.

Could you please test it once you have merged my changes?

Hope this helps, let us know

Thanks 

Jack

Comment (jtracey93) [reply avm/res/net-app/net-app-account/main.bicep:39]: Oh and this feature looks to be GA, i concur

Comment (eriqua) [reply avm/res/net-app/net-app-account/main.bicep]: Thanks @jtracey93 for chiming in and confirming!

@bobmclane999 regarding the screenshot you attached above, does the feature require to be registered or is that done by default? 
If the former @jtracey93 before merging, we should make sure the BRM target sub has that enabled. Also possibly we should add that prerequisite info to the README Notes section

Comment (eriqua) [reply avm/res/net-app/net-app-account/main.bicep]: Unresolving due to latest comment

Comment (eriqua) [reply avm/res/net-app/net-app-account/main.bicep]: Just checked that the feature is registered so we should be good to go from an environment prerequisites point of view. Resolving.

**Review by eriqua — COMMENTED** [review]
@fbinotto please have a look when you have a moment

File: .github/workflows/platform.toggle-avm-workflows.yml [hunk 2]
```diff
@@ -52,3 +52,4 @@ jobs:
 
           # Get the modified child resources
           Switch-WorkflowState @functionInput -Verbose
+# test
```
[the remark this answers is not in the document]
Comment (bobmclane999) [reply .github/workflows/platform.toggle-avm-workflows.yml]: removed. was added on a call with Jack to kick in to touch the workflows

**Comment by bobmclane999** [comment]
@microsoft-github-policy-service agree [company="Linklaters"]
@microsoft-github-policy-service agree
@microsoft-github-policy-service agree company="Microsoft"

**Comment by bobmclane999** [comment]
@microsoft-github-policy-service agree

**Comment by bobmclane999** [comment]
@microsoft-github-policy-service agree company="Linklaters"

**Comment by bobmclane999** [comment]
@microsoft-github-policy-service agree company=Linklaters
@microsoft-github-policy-service agree

**Review by fbinotto — APPROVED** [review]
LGTM

**Pull request merged**